xl-v1-multipart.go

```diff
@@ -631,9 +631,11 @@ func (xl xlObjects) CompleteMultipartUpload(bucket string, object string, upload
 
 	// Rename if an object already exists to temporary location.
 	uniqueID := getUUID()
-	err = xl.renameObject(bucket, object, minioMetaBucket, path.Join(tmpMetaPrefix, uniqueID))
-	if err != nil {
-		return "", toObjectErr(err, bucket, object)
+	if xl.isObject(bucket, object) {
+		err = xl.renameObject(bucket, object, minioMetaBucket, path.Join(tmpMetaPrefix, uniqueID))
```

Comment:
The problem was renameObject() was renaming the parent directory.

for ex: if object is `a/b/c/d` rename() on `a/b/` would succeed.

probably xl.isObject() can be moved inside xl.renameObject()